src/components/NamespaceList/ManageVisibilityModal.tsx

```diff
@@ -0,0 +1,212 @@
+        ),
+    );
+
+    return publicRoleBinding ? VisibilityOption.PUBLIC : VisibilityOption.PRIVATE;
+  }, [isLoading, roleBindingsError, roleBindings]);
+
+  // Initial form values based on current visibility
+  const initialValues: FormValues = {
+    visibility: currentVisibility,
+  };
+
+  const handleSubmit = async (values: FormValues, formikHelpers: FormikHelpers<FormValues>) => {
+    formikHelpers.setSubmitting(true);
+    setError(undefined);
+
+    try {
+      if (values.visibility === VisibilityOption.PUBLIC) {
+        // Create the role binding to make namespace public
+        const roleBinding: RoleBinding = {
+          apiVersion: 'rbac.authorization.k8s.io/v1',
+          kind: 'RoleBinding',
+          metadata: {
+            name: PUBLIC_ROLE_BINDING_NAME,
+            namespace: namespace.metadata.name,
+          },
+          roleRef: {
+            apiGroup: 'rbac.authorization.k8s.io',
+            kind: 'ClusterRole',
+            name: VIEWER_ROLE_NAME,
+          },
+          subjects: [
+            {
+              apiGroup: 'rbac.authorization.k8s.io',
+              kind: 'Group',
+              name: 'system:authenticated',
+            },
+          ],
+        };
+
+        await K8sQueryCreateResource({
+          model: RoleBindingModel,
+          resource: roleBinding,
+          queryOptions: {
+            ns: namespace.metadata.name,
+          },
+        });
+      } else {
+        // Delete the role binding to make namespace private
+        // Find the existing role binding from the current data
+        const publicRoleBinding = roleBindings?.find(
+          (rb) =>
+            rb.roleRef.name === VIEWER_ROLE_NAME &&
+            rb.subjects?.some(
+              (subject) => subject.kind === 'Group' && subject.name === 'system:authenticated',
+            ),
+        );
+
+        if (publicRoleBinding) {
+          await K8sQueryDeleteResource({
+            model: RoleBindingModel,
+            queryOptions: {
+              name: publicRoleBinding.metadata.name,
+              ns: namespace.metadata.name,
+            },
+          });
+        }
+      }
+
+      onClose();
+    } catch (err) {
+      setError(`Failed to save visibility setting: ${err.message || String(err)}`);
+    } finally {
+      formikHelpers.setSubmitting(false);
+    }
+  };
+
+  return (
+    <Formik initialValues={initialValues} onSubmit={handleSubmit} enableReinitialize>
+      {({ values, isSubmitting, handleSubmit: formikHandleSubmit }) => {
+        const hasChanges = values.visibility !== currentVisibility;
+
+        return (
+          <Modal
+            {...modalProps}
+            variant={ModalVariant.small}
+            title="Manage visibility"
+            data-testid="manage-visibility-modal"
+            description={
+              <>
+                Manage visibility for a namespace.{' '}
+                <ExternalLink href="https://placeholder-url.com">
```

Comment:
What is the plan for the href?  Normally, we do not recommend to merge PR with invalid links.